### PR TITLE
[Doc]Ctas supports index and order by

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 1]
```diff
@@ -45,11 +49,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
 
 | **Parameter**     | **Required** | **Description**                                              |
 | ----------------- | ------------ | ------------------------------------------------------------ |
-| column_name       | Yes          | The name of a column in the new table. You do not need to specify the data type for the column. StarRocks automatically specifies an appropriate data type for the column . StarRocks converts FLOAT and DOUBLE data into DECIMAL(38,9) data. StarRocks also converts CHAR, VARCHAR, and STRING data into VARCHAR(65533) data. |
+| column_name       | No          | The name of a column in the new table. You do not need to specify the data type for the column. StarRocks automatically specifies an appropriate data type for the column . StarRocks converts FLOAT and DOUBLE data into DECIMAL(38,9) data. StarRocks also converts CHAR, VARCHAR, and STRING data into VARCHAR(65533) data. |
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:52]: image
remove the sapce

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 2]
```diff
@@ -45,11 +49,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
...
+| partition_desc    | No           | The partitioning method of the new table. If you do not specify this parameter, by default, the new table has no partition. For more information about partitioning, see [CREATE TABLE](./CREATE_TABLE.md#partition_desc). |
+| distribution_desc | No           | The bucketing method of the new table. If you do not specify this parameter, the bucket column defaults to the column with the highest cardinality collected by the cost-based optimizer (CBO). The number of buckets defaults to 10. If the CBO does not collect information about the cardinality, the bucket column defaults to the first column in the new table. For more information about bucketing, see [CREATE TABLE](./CREATE_TABLE.md#distribution_desc). |
+| ORDER BY | No | Since v3.1.8, the sort key can be specified for the new table (which must be a Primary Key table). The sort key can be a combination of columns in any order. Note that, meanwhile, you must use `PRIMARY KEY (xxx)` to specify the new table as a Primary Key table.|
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:58]: the sort key can be specified for the new table (which must be a Primary Key table) -> a sort key can be specified for the new table if the table is a Primary Key table

Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:58]: Note that, meanwhile, you must use `PRIMARY KEY (xxx)` to specify the new table as a Primary Key table.
I guess you mean "A Primary Key table is a table for which `PRIMARY KEY (xxx)` is specified at table creation."

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 3]
```diff
@@ -17,11 +17,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
 
   ```SQL
   CREATE TABLE [IF NOT EXISTS] [database.]table_name
-  [(column_name [, column_name2, ...]]
+  [column_name [, column_name2, ...]]
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:20]: image
-> column_name1

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 4]
```diff
@@ -17,11 +17,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
...
+  [, index_definition1[, index_definition2, ...]]
   [key_desc]
   [COMMENT "table comment"]
   [partition_desc]
   [distribution_desc]
+  [ORDER BY (column_name,...)]
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:26]: I guess you mean "[ORDER BY (column_name [, column_name2, ...])]"?

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 5]
```diff
@@ -32,11 +34,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
   ```SQL
   SUBMIT [/*+ SET_VAR(key=value) */] TASK [[database.]<task_name>]AS
   CREATE TABLE [IF NOT EXISTS] [database.]table_name
-  [(column_name [, column_name2, ...]]
+  [column_name [, column_name2, ...]]
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:37]: See the comment on Line 20

File: docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 6]
```diff
@@ -32,11 +34,13 @@ You can submit an asynchronous CTAS task using [SUBMIT TASK](../data-manipulatio
...
+  [, index_definition1[, index_definition2, ...]]
   [key_desc]
   [COMMENT "table comment"]
   [partition_desc]
   [distribution_desc]
+  [ORDER BY (column_name,...)]
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:43]: See the comment on Line 26

File: docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 7]
```diff
@@ -17,11 +17,13 @@ CREATE TABLE AS SELECT（简称 CTAS）语句可用于同步或异步查询原
 
   ```SQL
   CREATE TABLE [IF NOT EXISTS] [database.]table_name
-  [(column_name [, column_name2, ...]]
+  [column_name [, column_name2, ...]]
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:20]: See the comment on Line 20 for the English doc

File: docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 8]
```diff
@@ -31,11 +33,13 @@ CREATE TABLE AS SELECT（简称 CTAS）语句可用于同步或异步查询原
   ```SQL
   SUBMIT [/*+ SET_VAR(key=value) */] TASK [[database.]<task_name>]AS
   CREATE TABLE [IF NOT EXISTS] [database.]table_name
-  [(column_name [, column_name2, ...]]
+  [column_name [, column_name2, ...]]
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:36]: See the comment on Line 20 for the English doc

File: docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 9]
```diff
@@ -31,11 +33,13 @@ CREATE TABLE AS SELECT（简称 CTAS）语句可用于同步或异步查询原
...
+  [, index_definition1[, index_definition2, ...]]
   [key_desc]
   [COMMENT "table comment"]
   [partition_desc]
   [distribution_desc]
+  [ORDER BY (column_name1,...)]
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:42]: See the comment on Line 26 for the English doc

File: docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md [hunk 10]
```diff
@@ -44,11 +48,13 @@ CREATE TABLE AS SELECT（简称 CTAS）语句可用于同步或异步查询原
...
 | key_desc          | 否       | 语法是 `key_type (<col_name1> [, <col_name2>, ...])`。<br />**参数**：<ul><li>`key_type`：新表的 Key 类型。有效值：`DUPLICATE KEY` 和 `PRIMARY KEY`。默认值：`DUPLICATE KEY`。</li><li> `col_name`：组成 Key 的列。</li></ul>|
 | COMMENT           | 否       | 新表注释。                                                   |
-| partition_desc    | 否       | 新表的分区方式。如不指定该参数，则默认新表为无分区。更多有关分区的设置，参见 CREATE TABLE。 |
-| distribution_desc | 否       | 新表的分桶方式。如不指定该参数，则默认新表的分桶列为使用 CBO 优化器采集的统计信息中基数最高的列，且分桶数量默认为 10。如果 CBO 优化器没有采集基数信息，则默认新表的第一列为分桶列。更多有关分桶的设置，参见 CREATE TABLE。 |
+| partition_desc    | 否       | 新表的分区方式。如不指定该参数，则默认新表为无分区。更多有关分区的设置，参见 [CREATE TABLE](./CREATE_TABLE.md#partition_desc)。 |
+| distribution_desc | 否       | 新表的分桶方式。如不指定该参数，则默认新表的分桶列为使用 CBO 优化器采集的统计信息中基数最高的列，且分桶数量默认为 10。如果 CBO 优化器没有采集基数信息，则默认新表的第一列为分桶列。更多有关分桶的设置，参见 [CREATE TABLE](./CREATE_TABLE.md#distribution_desc)。 |
+| ORDER BY| 否       |自 3.1.8 开始支持为新表（必须是主键表）指定排序键，可以为任意列的排列组合。注意此时您必须使用  `PRIMARY KEY (xxx)` 指定新表为主键表。|
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-definition/CREATE_TABLE_AS_SELECT.md:57]: See the comment for the same part in the English doc

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**